packages/syft/src/syft/core/tensor/lazy_repeat_array.py

```diff
@@ -37,7 +37,9 @@ class lazyrepeatarray:
 
     __attr_allowlist__ = ["data", "shape"]
 
-    def __init__(self, data: np.ndarray, shape: Tuple[int, ...]) -> None:
+    def __init__(
+        self, data: np.ndarray, shape: Tuple[int, ...], data_type: str = "none"
```

Comment:
Perhaps we should extract these stringly typed values to some file level CONSTANTS or an Enum and if there is a use case without min / max, make it Optional where `None` represents no min or max flatten operation. If more operations vary in their treatment this could be moved further up the chain by creating a Min and Max subclass which implement the correct handling leaving the general case to the base class?

Reply:
(Great catch!) I'm not sure it's worth the effort to make min and max subclasses- I don't see lazyrepeatarray being used for anything else besides storing min and max values in DP tensors, so it might be best to keep things simple for now?

I actually didn't know about Enums in Python- looks cool and I definitely think they can fit in here :D (based on my understanding from https://stackoverflow.com/questions/37601644/python-whats-the-enum-type-good-for)

Reply:
Good suggestion, @madhavajay. I've used enums in TypeScript (https://www.typescriptlang.org/docs/handbook/enums.html) but didn't even know they were available in Python!

Reply:
@madhavajay @IshanMi I've added a TODO to create an enum in a separate PR for now.